packages/git/src/browser/diff/git-diff-widget.ts

```diff
@@ -125,74 +110,95 @@ export class GitDiffWidget extends GitBaseWidget implements StatefulWidget {
     }
 
     protected renderDiffListHeader(): h.Child {
-        let fileDiv: h.Child = '';
-        if (this.viewModel.gitDiffFile && !this.viewModel.title) {
-            const iconSpan = h.span({ className: this.viewModel.gitDiffFile.icon + ' file-icon' });
-            const nameSpan = h.span({ className: 'name' }, this.viewModel.gitDiffFile.label + ' ');
-            const pathSpan = h.span({ className: 'path' }, this.viewModel.gitDiffFile.path);
-            const compareDiv = h.span({}, 'Compare ');
-            fileDiv = h.div({ className: "gitItem diff-file" }, h.div({ className: "noWrapInfo" }, compareDiv, iconSpan, nameSpan, pathSpan));
-            const withSpan = h.span({ className: 'row-title' }, 'with ');
-            const fromDiv =
-                this.viewModel.fromRevision && typeof this.viewModel.fromRevision !== 'number' ?
-                    h.div({ className: "revision noWrapInfo" }, withSpan, this.viewModel.fromRevision.toString()) :
-                    'previous revision';
-            return h.div({ className: "commitishBar" }, fileDiv, fromDiv);
-        } else {
-            const header = this.viewModel.title ? h.div({ className: 'git-diff-header' }, this.viewModel.title) : '';
-            return h.div({ className: "commitishBar" }, header);
+        const elements = [];
+        if (this.options.uri) {
+            const path = this.relativePath(this.options.uri);
+            if (path.length > 0) {
+                elements.push(h.div({ className: 'header-row' },
+                    h.div({ className: 'theia-header' }, 'path:'),
+                    h.div({ className: 'header-value' }, '/' + path)));
+            }
+        }
+        if (this.fromRevision) {
+            let revision;
+            if (typeof this.fromRevision === 'string') {
+                revision = h.div({ className: 'header-value' }, this.fromRevision);
+            } else {
+                revision = h.div({ className: 'header-value' }, (this.toRevision || 'HEAD') + '~' + this.fromRevision);
```

Comment:
I find it strange that the UI has the `toRevision`~`fromRevisionn` conventions here, while the [URI](https://github.com/theia-ide/theia/pull/1092/commits/597b99cdbead969d0cadcc498061dda8d15c5fb9#diff-2f815dd2c1eac9fc05fea815e29fcc32R167) is built the other way around: `from` \~`to`.

Reply:
It is not a range in this case. fromRevision is a number here so we just want to show the relative revision (e.g. HEAD~4).